bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/action/AstroActions.java

```diff
@@ -86,6 +88,25 @@ public void setThingHandler(@Nullable ThingHandler handler) {
         return null;
     }
 
+    @RuleAction(label = "get the total sun radiation", description = "Get the total sun radiation for a given time.")
+    public @Nullable @ActionOutput(name = "getTotalRadiation", label = "Total Radiation", type = "org.openhab.core.library.types.QuantityType<org.openhab.core.library.dimension.Intensity>") QuantityType<Intensity> getTotalRadiation(
+            @ActionInput(name = "date", label = "Date", required = false, description = "Considered date") @Nullable ZonedDateTime date) {
+        logger.debug("Astro action 'getTotalRadiation' called");
+        AstroThingHandler theHandler = this.handler;
+        if (theHandler != null) {
+            if (theHandler instanceof SunHandler) {
+                SunHandler handler = (SunHandler) theHandler;
+                Radiation radiation = handler.getRadiationAt(date != null ? date : ZonedDateTime.now());
```

Comment:
```suggestion
            if (theHandler instanceof SunHandler sunHandler) {
                Radiation radiation = sunHandler.getRadiationAt(date != null ? date : ZonedDateTime.now());
```

Reply:
As this is a java 17 feature this suggested change is not 3.4.x compatible as stated in the start post. Not a big deal though.

Reply:
Correct, as an enhancement this is not being considered for 3.4.x anyway.